plugins/woocommerce/client/legacy/js/admin/settings.js

```diff
@@ -47,6 +47,17 @@
 			} )
 			.trigger( 'change' );
 
+		// Customer stock notifications
+		$( 'input#woocommerce_customer_stock_notifications_require_double_opt_in' )
+			.on( 'change', function () {
+				if ( $( this ).is( ':checked' ) ) {
+					$( this ).closest( 'tr' ).next( 'tr' ).show();
+				} else {
+					$( this ).closest( 'tr' ).next( 'tr' ).hide();
+				}
+			} )
+			.trigger( 'change' );
+
```

Comment:
Woo settings API already has support for 'show/hide' based on an other field's value, however, the API only works when both dependent fields are checkboxes.

In our case, one field is and the other is , so we needed to add a JS snippet to handle our setting dependencies.